core/templates/components/common-layout-directives/common-elements/loading-dots.component.ts

```diff
@@ -16,18 +16,17 @@
  * @fileoverview Directive for displaying animated loading dots.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
 
-angular.module('oppia').directive('loadingDots', [
-  'UrlInterpolationService', function(UrlInterpolationService) {
-    return {
-      restrict: 'E',
-      scope: {},
-      bindToController: {},
-      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
-        '/components/common-layout-directives/common-elements/' +
-        'loading-dots.directive.html'),
-      controllerAs: '$ctrl',
-      controller: [function() {}]
-    };
-  }]);
+
+@Component({
+  selector: 'loading-dots',
+  template: require('./loading-dots.component.html')
```

Comment:
Hi @bansalnitish, last week I tried to test an updated directive but some errors came up. Can you please test this specific component? The error was related to the template property must receive only strings (require is not allowed on unit tests).

I think we should resolve this problem before migrating a lot of directives to avoid blocking the progress of frontend tests.

Reply:
Hi Mariana, The problem is fixed now. Please look at the changes.

Reply:
Hi Nitish, did you test this change in the application? I've been trying to change it on donate page but it's breaking the application